velox/vector/SelectivityVector.h

```diff
@@ -156,7 +156,7 @@ class SelectivityVector {
     bits::fillBits(bits_.data(), 0, size_, false);
     begin_ = 0;
     end_ = 0;
-    allSelected_ = false;
+    allSelected_.reset();
```

Comment:
This is not correct.  `allSelected_` has 3 states, `nullopt`, `true`, `false`, here we should set it to `false`, not `nullopt`

Reply:
Yes. I intended to keep this logic but set it to `false` directly will cause the issue as mentioned, even with a check like `if (allSelected_.has_value() && *allSelected_)` first. The variable is used in `isAllSelected()` and `countSelected()`, which can also give correct return if it doesn't have value, so using `allSelected_.reset()` here is acceptable? This error is kind of code sensitive and I didn't search out a formal resolution for it, any other suggestion from you?

Reply:
This is an optimization and we don't want to lose it.  The issue apparently is a bug in compiler and should be worked around by some compiler flag/hints